### DISCUSSION TITLE
Setup Herd Mail service for local Wordpress

### DISCUSSION THREAD

**Discussion by philbraun** [beyondcode/herd-community#1170] [General]
Using Herd as a replacement for MAMP Pro / Valet / localWP – strictly WordPress local development.
While this (still) runs rather nicely, I'm not sure I understand how the Mail service is supposed to work. My Wordpress sites naturally all are without any .env files, as there's no need for any the Laravel stuff. I tried manually adding an .env file with just the mail setup to a site and created a Laravel project site, but never got the Mail window to display anything.

In the Herd Wordpress docs, I found the bit about «The quickest way to set up mails in Herd Pro is by defining a mailer in the functions.php. Simply paste the following snippet to the end of your functions.php to receive emails in Herd.»

So first of all, which functions.php are we talking about here: the one in every WP Theme? And if so, this means I'll have to add this bit to every Theme I ever intend to use or even just test locally, and then again remove it before uploading to any public staging server or to any live site. That seems a somewhat clunky approach to me. 

So my question: is there another way to get the Mail service running, e.g. with a per-site .env file, and what's the minimum that has to be in such an .env file for it to be recognized and acted on by Herd?

**Comment by vinkla** [comment]
> So first of all, which functions.php are we talking about here: the one in every WP Theme?

Yes, you need to add this to every theme, or you can build and distribute a plugin.

> So my question: is there another way to get the Mail service running, e.g. with a per-site .env file, and what's the minimum that has to be in such an .env file for it to be recognized and acted on by Herd?

I recommend incorporating this into your `functions.php` file and retrieving the credentials from your environment, as you have already stated. Below you'll find links to how I've personally implemented this in WordPress:

- [`functions.php`](https://github.com/vinkla/wordplate/blob/49a5da712f4f8b3e60f2472e0d106e83b299fe03/public/themes/wordplate/functions.php#L98-L114)
- [`.env`](https://github.com/vinkla/wordplate/blob/49a5da712f4f8b3e60f2472e0d106e83b299fe03/.env.example#L11-L17)
